apps/lvgl_scanner/src/main.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "sysinit/sysinit.h"
+#include "os/os.h"
+#include "console/console.h"
+#include "host/ble_hs.h"
+#include "log/log.h"
+#include "lvgl.h"
+
+#define MAX_ENTRIES         30
+#define ENTRY_ID_NONE       (MAX_ENTRIES + 1)
+
+lv_obj_t *entries_list;
+
+struct list_entry_data {
+    int8_t last_rssi;
+    uint8_t addr_val[6];
+} entry_data[MAX_ENTRIES];
+
+static void
+sort_entries(void)
+{
+    lv_obj_t *entry;
+    int8_t rssi1;
+    int8_t rssi2;
+    uint8_t i;
+    uint8_t j;
+    uint8_t n = lv_obj_get_child_cnt(entries_list);
+
+    for (i = 0; i < n - 1; i++) {
+        for (j = 0; j < n - i - 1; j++) {
+            rssi1 = entry_data[j].last_rssi;
+            rssi2 = entry_data[j + 1].last_rssi;
+
+            entry = lv_obj_get_child(entries_list, j + 1);
+            if (rssi1 < rssi2) {
+                struct list_entry_data help = entry_data[j + 1];
+                entry_data[j + 1] = entry_data[j];
+                entry_data[j] = help;
+                lv_obj_move_to_index(entry, j);
```

Comment:
you can add pointer to object to `entry_data` struct, then simply sort entries and move objects to appropriate positions after sorting is done. this way you don't need to move them multiple times during sorting